pkg/controller/sparkapplication/controller.go

```diff
@@ -79,6 +79,8 @@ type Controller struct {
 	ingressClassName  string
 	batchSchedulerMgr *batchscheduler.SchedulerManager
 	enableUIService   bool
+	enableIngressTLS bool
```

Comment:
Nit: these two and the two fields above related to ingress can be grouped together.

Reply:
I thought it makes sense to split the configuration because you might want to enableUI service but still let the user provide all other configurations. The `enableIngressTLS` is meant to be turned on only if you have a cluster wide provider and want simplify users' lives.